### PR TITLE
feat: icky-810 multiship integration test and fixes

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/checkout/DetailsStep/DetailsStep.spec.tsx [hunk 1]
```diff
@@ -3,47 +3,122 @@
...
+        const emailInput = screen.getByRole('textbox', { name: /your-email/i })
+
+        expect(emailTitle).toBeVisible()
+        expect(emailInput).toBeVisible()
+        expect(
+          screen.queryByRole('button', { name: /sign-into-your-account/i })
+        ).not.toBeInTheDocument()
+        expect(screen.queryByText(/or-fill-the-details-below/i)).not.toBeInTheDocument()
+      })
+    })
```
Comment (kibo-sushant) [inline components/checkout/DetailsStep/DetailsStep.spec.tsx:74]: should render component
- should verify whether all the items are rendered or not (features section and email input)

should not render sign-into-your-account button
- should verify items those should not be rendered

File: components/checkout/MultiShippingStep/MultiShippingStep.spec.tsx [hunk 2]
```diff
@@ -126,237 +139,344 @@ jest.mock('../../common/ProductItemWithAddressList/ProductItemWithAddressList',
...
   __esModule: true,
-  default: () => <div data-testid="shipping-groups-with-method"></div>,
+  default: ({
+    onUpdateCheckoutShippingMethod,
+  }: {
+    onUpdateCheckoutShippingMethod: (params: {
+      shippingMethodGroup: CheckoutGroupRates
+      shippingMethodCode: string
+    }) => void
+  }) => (
```
Comment (kibo-sushant) [inline components/checkout/MultiShippingStep/MultiShippingStep.spec.tsx]: To make code readable separate type/interface and use it

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 3]
```diff
@@ -1,48 +1,273 @@
...
+  __esModule: true,
+  default: ({
+    id,
+    selected,
+    onShippingMethodChange,
+  }: {
+    id: number
+    selected: string
+    onShippingMethodChange: (shippingMethodCode: string) => void
+  }) => (
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: To make the code readable separate the type/interface and apply it
Destructor params to next line

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 4]
```diff
@@ -1,48 +1,273 @@
...
+  __esModule: true,
+  default: ({
+    id,
+    selected,
+    handleRadioChange,
+  }: {
+    id: number
+    selected: string
+    handleRadioChange: (addressId: string) => void
+  }) => (
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: To make the code readable separate the type/interface and apply it
Destructor params to next line

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 5]
```diff
@@ -1,48 +1,273 @@
...
+
+jest.mock('../../common/AddressForm/AddressForm', () => ({
+  __esModule: true,
+  default: ({
+    onSaveAddress,
+    onFormStatusChange,
+  }: {
+    onSaveAddress: (address: Address) => void
+    onFormStatusChange: (isValid: boolean) => void
+  }) => (
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: To make the code readable separate the type/interface and apply it
Destructor params to next line

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 6]
```diff
@@ -1,48 +1,273 @@
...
+                cityOrTown: 'city',
+                stateOrProvince: 'state',
+                postalOrZipCode: '23423',
+                countryCode: 'US',
+                isValidated: false,
+              },
+            },
+            isAddressValid: true,
+          })
+        }
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: To make code readable assign values to one const and use it here

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 7]
```diff
@@ -1,48 +1,273 @@
...
+jest.mock('../../checkout/ShippingMethod/ShippingMethod', () => ({
+  __esModule: true,
+  default: ({
+    id,
+    selected,
+    onShippingMethodChange,
+  }: {
+    id: number
+    selected: string
+    onShippingMethodChange: (shippingMethodCode: string) => void
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: missing onStoreLocatorClick

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 8]
```diff
@@ -1,48 +1,273 @@
...
+      <button onClick={() => handleRadioChange(String(id))}>handleAddressSelect</button>
+    </div>
+  ),
+}))
+
+jest.mock('../../common/AddressForm/AddressForm', () => ({
+  __esModule: true,
+  default: ({
+    onSaveAddress,
+    onFormStatusChange,
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: validateForm is missing
should accept children and render it

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx [hunk 9]
```diff
@@ -1,48 +1,273 @@
...
+        name: /handleSaveShippingMethod/,
+      })[0]
+
+      await user.click(firstShippingMethod)
 
-    const shippingHeading = screen.getAllByRole('heading', {
-      name: /shipping/i,
+      await waitFor(() => {
+        expect(scrollIntoViewMock).toBeCalled()
+      })
```
Comment (kibo-sushant) [inline components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx:279]: **Test case names are not meaningfull**
should display shipping address(saved in account) radio buttons and select a radio button to select Shipping Methods
should select a shipping address radio button and shippingMethod should be selected

**Missing Testcases**
should able to save new address when user fills the address form and clicks on Save button

**Stories do not display Shipping Methods**

File: components/checkout/PaymentStep/PaymentStep.stories.tsx [hunk 10]
```diff
@@ -79,19 +79,19 @@ WithSavedPayment.args = {
...
-      }),
-    },
-  },
-}
+//       contacts: graphql.query('getUserAddresses', (_req, res, ctx) => {
+//         return res(ctx.data(userAddressMock))
+//       }),
+//     },
+//   },
+// }
```
Comment (kibo-sushant) [inline components/checkout/PaymentStep/PaymentStep.stories.tsx]: Remove unused imports customerAccountCardsMock, userAddressMock

There no is difference between following stories
 WithoutSavedPayment
 WithSavedPayment
 MultiShip

File: components/checkout/MultiShippingStep/MultiShippingStep.tsx [hunk 11]
```diff
@@ -29,6 +30,7 @@ import type {
   CustomerContact,
   Checkout,
   CheckoutGroupRates,
+  CustomerContactCollection,
```
Comment (kibo-sushant) [inline components/checkout/MultiShippingStep/MultiShippingStep.tsx:32]: unused imports MultiShipAddress, CrOrderItem

File: components/checkout/MultiShippingStep/MultiShippingStep.tsx [hunk 12]
```diff
@@ -329,12 +342,21 @@ const MultiShippingStep = (props: MultiShippingStepProps) => {
...
   }, [isMultiShipPaymentStepValid, checkout])
 
+  useEffect(() => {
+    if (
+      (checkout?.groupings?.filter((group) => group?.fulfillmentMethod === FulfillmentOptions.SHIP)
+        .length as number) < 2
+    ) {
+      setShippingOption(shipToHome)
+    }
+  }, [checkout?.groupings?.length])
```
Comment (kibo-sushant) [inline components/checkout/MultiShippingStep/MultiShippingStep.tsx:351]: To make the code readable assign result to const and use the const in if
ex.
 const shipItems = checkout?.groupings?.filter((group) => group?.fulfillmentMethod === FulfillmentOptions.SHIP
 if(shipItems.length < 2) setShippingOption(shipToHome)

File: components/common/PasswordValidation/PasswordValidation.spec.tsx [hunk 13]
```diff
@@ -3,8 +3,8 @@ import React from 'react'
 import { composeStories } from '@storybook/testing-react'
 import { render, screen } from '@testing-library/react'
 
-import theme from '../../../styles/theme'
 import * as stories from './PasswordValidation.stories'
+import theme from '../../../styles/theme'
```
Comment (kibo-sushant) [inline components/common/PasswordValidation/PasswordValidation.spec.tsx]: use @/styles/theme

File: components/common/Price/Price.spec.js [hunk 14]
```diff
@@ -3,8 +3,8 @@ import React from 'react'
 import { composeStories } from '@storybook/testing-react'
 import { render, screen } from '@testing-library/react'
 
-import theme from '../../../styles/theme'
 import * as stories from './Price.stories' // import all stories from the stories file
+import theme from '../../../styles/theme'
```
Comment (kibo-sushant) [inline components/common/Price/Price.spec.js]: use @/styles/theme

File: components/core/ImageGallery/ImageGallery.stories.tsx [hunk 15]
```diff
@@ -2,8 +2,8 @@ import React from 'react'
 
 import { ComponentStory, ComponentMeta } from '@storybook/react'
 
-import { imageGalleryData } from '../../../__mocks__/stories/imageGalleryDataMock'
 import ImageGallery from './ImageGallery'
+import { imageGalleryData } from '../../../__mocks__/stories/imageGalleryDataMock'
```
Comment (kibo-sushant) [inline components/core/ImageGallery/ImageGallery.stories.tsx]: use @/__mocks__/stories/imageGalleryDataMock

File: components/core/ImageGallery/ImageGallery.spec.js [hunk 16]
```diff
@@ -6,8 +6,8 @@ import '@testing-library/jest-dom'
 import userEvent from '@testing-library/user-event'
 import { act } from 'react-dom/test-utils'
 
-import { imageGalleryData } from '../../../__mocks__/stories/imageGalleryDataMock'
 import * as stories from './ImageGallery.stories'
+import { imageGalleryData } from '../../../__mocks__/stories/imageGalleryDataMock'
```
Comment (kibo-sushant) [inline components/core/ImageGallery/ImageGallery.spec.js]: use @/__mocks__/stories/imageGalleryDataMock

File: components/layout/RegisterAccount/RegisterAccountDialog/RegistAccountDialog.spec.tsx [hunk 17]
```diff
@@ -4,8 +4,8 @@ import { composeStories } from '@storybook/testing-react'
 import { render, screen } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
 
-import { KiboDialogProps } from '../../../common/KiboDialog/KiboDialog'
 import * as stories from './RegisterAccountDialog.stories' // import all stories from the stories file
+import { KiboDialogProps } from '../../../common/KiboDialog/KiboDialog'
```
Comment (kibo-sushant) [inline components/layout/RegisterAccount/RegisterAccountDialog/RegistAccountDialog.spec.tsx]: use @/components/common/KiboDialog/KiboDialog

File: components/product/ProductOptionSelect/ProductOptionSelect.spec.js [hunk 18]
```diff
@@ -3,8 +3,8 @@ import { render, screen, within, fireEvent } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
 
 import '@testing-library/jest-dom'
-import { productOptionSelectValuesMock } from '../../../__mocks__/stories/productOptionSelectMock'
 import * as stories from './ProductOptionSelect.stories' // import all stories from the stories file
+import { productOptionSelectValuesMock } from '../../../__mocks__/stories/productOptionSelectMock'
```
Comment (kibo-sushant) [inline components/product/ProductOptionSelect/ProductOptionSelect.spec.js]: use @/__mocks__/stories/productOptionSelectMock

File: __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx [hunk 19]
```diff
@@ -1,138 +1,1175 @@
...
+  CrOrderItem,
+  CrPayment,
+  CrProduct,
+  CustomerContact,
+} from '@/lib/gql/types'
 
 const scrollIntoViewMock = jest.fn()
 window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock
 
+jest.mock('next/router', () => require('next-router-mock'))
```
Comment (kibo-sushant) [inline __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx]: No need of this, as its already defined in jest.setup

File: __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx [hunk 20]
```diff
@@ -1,138 +1,1175 @@
...
+          })
+
+          checkoutData.mock = await handleDetailsStep(user, checkoutData.mock)
+
+          checkoutData.mock = await handleMultiShipToHomeItems(user, checkoutData.mock)
+
+          checkoutData.mock = await handlePaymentStep(user, checkoutData.mock)
+
+          await handleReviewStep(user, checkoutData.mock)
+        })
```
Comment (kibo-sushant) [inline __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx]: **1) Define all the static data at top and use it whenever is required with server.use 
  (To make the code readable)**
ex.
  const getMultiShipCheckoutData = []
  const getCheckoutShippingMethods = []
  ...
 
  server.use(
    graphql.query('getMultiShipCheckout', (_req, res, ctx) => res( ctx.data({...getMultiShipCheckoutData }) ))
  )

**2) To many nested describe blocks.**
   Are we handling For UnAuthenticated user? If not then no need of  describe('Authenticated user')

 describe('[integration] MultiShipCheckoutTemplate', () => {
  describe('Authenticated user', () => {
    describe("user doesn't have any saved addresses", () => {
      describe('checking out for one single ship to home and one pickup in store item', () => {
	 it('should handle the checkout flow', async () => {
      describe('checking out for more than one single ship to home', () => {
	 it('should handle checkout flow', async () => {

**3) should convert function to corresponding testcases** @kibo-kevinwatts , your thoughts please.
ex. 
 should activate Shipping step when user fills the Details Step and Clicks on go-to-shipping button
 should activate Payment step when user fills the Shipping Step and Clicks on go-to-payment button
 ...
 ...

**Review by kibo-sushant — CHANGES_REQUESTED** [review]

File: components/checkout/OrderReview/OrderReview.tsx [hunk 21]
```diff
@@ -257,10 +243,11 @@ const OrderReview = (props: OrderReviewProps) => {
                 display="inline"
                 pt={1}
                 key={`${paymentMethod?.cardNumberPartOrMask}-${paymentMethod?.expiry}`}
+                data-testid="payment-info"
               >
                 <Typography variant="body1">{paymentMethod?.cardType}</Typography>
                 <Typography variant="body1">{paymentMethod?.cardNumberPartOrMask}</Typography>
-                <Typography variant="body1">{paymentMethod?.expiry} XXX</Typography>
+                <Typography variant="body1">{paymentMethod?.expiry}</Typography>
```
Comment (kibo-sushant) [inline components/checkout/OrderReview/OrderReview.tsx:241]: unused variable 
 const storeLocations = storeLocationGetters.getLocations(locations as Maybe<Location>[])

**Review by kibo-sushant — CHANGES_REQUESTED** [review]

File: components/checkout/PaymentStep/PaymentStep.stories.tsx (hunk 10 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/PaymentStep/PaymentStep.stories.tsx]: Done

File: components/checkout/MultiShippingStep/MultiShippingStep.tsx (hunk 11 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/MultiShippingStep/MultiShippingStep.tsx:32]: Done

File: components/checkout/MultiShippingStep/MultiShippingStep.tsx (hunk 12 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/MultiShippingStep/MultiShippingStep.tsx:351]: Done

File: components/common/PasswordValidation/PasswordValidation.spec.tsx (hunk 13 above)
Comment (kibo-chandradeeptalaha) [reply components/common/PasswordValidation/PasswordValidation.spec.tsx]: Done

File: components/common/Price/Price.spec.js (hunk 14 above)
Comment (kibo-chandradeeptalaha) [reply components/common/Price/Price.spec.js]: Done

File: components/core/ImageGallery/ImageGallery.stories.tsx (hunk 15 above)
Comment (kibo-chandradeeptalaha) [reply components/core/ImageGallery/ImageGallery.stories.tsx]: Done

File: components/core/ImageGallery/ImageGallery.spec.js (hunk 16 above)
Comment (kibo-chandradeeptalaha) [reply components/core/ImageGallery/ImageGallery.spec.js]: Done

File: components/layout/RegisterAccount/RegisterAccountDialog/RegistAccountDialog.spec.tsx (hunk 17 above)
Comment (kibo-chandradeeptalaha) [reply components/layout/RegisterAccount/RegisterAccountDialog/RegistAccountDialog.spec.tsx]: Done

File: components/product/ProductOptionSelect/ProductOptionSelect.spec.js (hunk 18 above)
Comment (kibo-chandradeeptalaha) [reply components/product/ProductOptionSelect/ProductOptionSelect.spec.js]: Done

File: __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx (hunk 19 above)
Comment (kibo-chandradeeptalaha) [reply __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx]: Done

File: __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx (hunk 20 above)
Comment (kibo-chandradeeptalaha) [reply __test__/e2e/integration/checkout/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.spec.tsx]: 1. Done.
2. Done

File: components/checkout/MultiShippingStep/MultiShippingStep.spec.tsx (hunk 2 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/MultiShippingStep/MultiShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 3 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 4 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 5 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 8 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 9 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx:279]: Test case name changed.

Missing test case name changed: Handled in Integration test.

File: components/checkout/DetailsStep/DetailsStep.spec.tsx (hunk 1 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/DetailsStep/DetailsStep.spec.tsx:74]: Already covered

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 6 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: Done

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 7 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx]: feature not implemented yet. Need separate ticket for this.

File: components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx (hunk 9 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/StandardShippingStep/StandardShippingStep.spec.tsx:279]: @kibo-sushant  Stories do not display Shipping Methods // any specific scenario ?

File: components/checkout/OrderReview/OrderReview.tsx (hunk 21 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/OrderReview/OrderReview.tsx:241]: Removed

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**